core/lib/l1_contract_interface/src/i_executor/structures/commit_batch_info.rs

```diff
@@ -208,6 +209,13 @@ impl Tokenizable for CommitBatchInfo<'_> {
                     vec![PUBDATA_SOURCE_BLOBS]
                 }
 
+                (L1BatchCommitmentMode::Rollup, PubdataDA::Custom) => {
+                    panic!("Custom pubdata DA is incompatible with Rollup mode")
```

Comment:
Don't quite like this panic (instead of e.g. returning an error). Not critical, but maybe worth investigating whether this can be defined as an inherent `CommitBatchInfo` method.

Reply:
I agree that it doesn't look nice here, but the alternative would be to update the `Tokenizable` trait and change the `fn into_token(self) -> Token` to `fn into_token(self) -> Result<Token, Error>`
but that requires changing a lot of code that depends on it, so I'm not sure whether it's worth it

or did you mean some other approach?

Reply:
Changing the trait is certainly not worth it. I meant defining an inherent method it `CommitBatchInfo` instead of implementing the `Tokenizable` trait for it:

```rust
impl CommitBatchInfo<'_> {
    pub fn try_into_token(self) -> Result<Token, _> {
        // ...
    }
}
```

Reply:
I'll mark this as resolved to be able to merge the PR, but I will keep this in mind for the follow-up PR, and try the suggested solution